main.tf

```
@@ -25,6 +25,16 @@ resource "datadog_integration_aws" "integration" {
   account_specific_namespace_rules = var.account_specific_namespace_rules
 }
 
+module "kinesis" {
+  count   = var.kinesis_enabled ? 1 : 0
```

The module should respect the enabled flag on it which is passed by the context but also by `var.kinesis_enabled`

```suggestion
  enabled = local.enabled && var.kinesis_enabled ? 1 : 0
```